File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer,Integer>map=new HashMap<>();
        for(int i:arr){
            if(!map.containsKey(i)){
                map.put(i,1);
            }
            else{
                map.put(i,map.get(i)+1);
            }
        }
        
        HashSet<Integer>set=new HashSet<>(map.values());
        int n=map.size();
        int m=set.size();
        
        if(n==m){
            return true;
        }
        
        return false;
    }
}

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.kt
class Solution {
fun uniqueOccurrences(arr: IntArray): Boolean {
    val map = mutableMapOf<Int, Int>()
    arr.forEach {
        map[it] = map.getOrDefault(it, 0) + 1
    }
    return map.size == map.values.toSet().size

}
}